### PR TITLE
Removes use of /page/references

### PR DESCRIPTION
**Pull request by sharvaniharan**
Removing final bits of /page/references

### CODE REVIEW

File: app/src/main/java/org/wikipedia/page/references/PageReferences.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+@SuppressWarnings("unused,NullableProblems")
+public class PageReferences {
+    @Nullable @SerializedName("selectedIndex") private int selectedIndex;
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/references/PageReferences.java]: The `@SerializedName` annotation is not necessary when the name of the field is the same.

File: app/src/main/java/org/wikipedia/dataclient/RestService.java [hunk 2]
```diff
@@ -116,12 +116,6 @@
...
     @NonNull Observable<FeedAvailability> getFeedAvailability();
 
-    @GET("page/references/{title}/{revision}")
-    @NonNull Observable<References> getReferences(@NonNull @Path("title") String title,
-                                                  @Path("revision") long revision,
-                                                  @Nullable @Header(OfflineCacheInterceptor.LANG_HEADER) String langHeader,
-                                                  @Nullable @Header(OfflineCacheInterceptor.TITLE_HEADER) String titleHeader);
-
     @GET("page/references/{title}/{revision}")
     @NonNull Observable<Response<References>> getReferencesResponse(@Path("title") String title,
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/dataclient/RestService.java]: Can we get rid of `getReferencesResponse`, too?

File: app/src/main/java/org/wikipedia/page/references/PageReferences.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        @Nullable @SerializedName("html") private String content;
+        @Nullable @SerializedName("href") private String href;
+        private String text;
+
+        @NonNull
+        public String getContent() {
+            return StringUtils.defaultString(content);
+        }
+
+        public void setText(@NonNull String text) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/references/PageReferences.java]: Is this used?

File: app/src/main/java/org/wikipedia/page/references/PageReferences.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            this.text = text;
+        }
+
+        @NonNull
+        public String getText() {
+            return text;
+        }
+
+        @Nullable
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/references/PageReferences.java]: Why not `@NonNull`?

File: app/src/main/java/org/wikipedia/page/references/PageReferences.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            return StringUtils.defaultString(content);
+        }
+
+        public void setText(@NonNull String text) {
+            this.text = text;
+        }
+
+        @NonNull
+        public String getText() {
+            return text;
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/references/PageReferences.java]: Shouldn't we use `StringUtils.defaultString()` here, like elsewhere?

File: app/src/main/java/org/wikipedia/page/references/PageReferences.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    @Nullable @SerializedName("selectedIndex") private int selectedIndex;
+
+    @Nullable private String tid;
+    @Nullable @SerializedName("referencesGroup") private Reference[] referencesGroup;
+
+    public int getSelectedIndex() {
+        return selectedIndex;
+    }
+
+    @Nullable public List<Reference> getReferencesGroup() {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/page/references/PageReferences.java]: Why not `@NonNull`?

**Comment by sharvaniharan** [comment]
Done @dbrant

**Pull request merged**